ipatests/test_integration/test_acme.py

```diff
@@ -252,3 +264,72 @@ def test_disable_acme_service(self):
             ['curl', '--fail', self.acme_server],
             ok_returncode=22,
         )
+
+    def test_centralize_acme_disable(self):
+        """Test if ACME disable on replica if disabled on master"""
+        result = self.replicas[0].run_command(['ipa-acme-manage', 'status'])
+        assert 'disabled' in result.stdout_text
+
+
+@pytest.mark.skipif(not cainstance.minimum_acme_support(),
+                    reason="does not provide ACME")
+class TestACMECALess(IntegrationTest):
+    """Test to check the CA less replica setup"""
+    num_replicas = 1
+    num_clients = 1
+
+    #########################################
+    #                TODO                   #
+    # refactor install after PR#5147 merged #
+    #########################################
+    @classmethod
+    def install(cls, mh):
+        # cache the acme service uri
+        acme_host = f'{IPA_CA_RECORD}.{cls.master.domain.name}'
+        cls.acme_server = f'https://{acme_host}/acme/directory'
+
+        # install packages before client install in case of IPA DNS problems
+        if not skip_certbot_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'certbot'])
+        if not skip_mod_md_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'mod_md'])
+
+        tasks.install_master(cls.master, setup_dns=True)
+
+        tasks.install_client(cls.master, cls.clients[0])
+        tasks.config_host_resolvconf_with_master_data(
+            cls.master, cls.clients[0]
+        )
+        tasks.install_replica(cls.master, cls.replicas[0], setup_ca=False)
+        tasks.config_host_resolvconf_with_master_data(
+            cls.master, cls.replicas[0]
+        )
+
+    def test_cafull_centralize_acme_enable(self):
+        """Test if ACME is enabled after CA-less to CA-full deployment
+
+        Deployment where one server is deployed as CA-less, when converted
+        to CA full, should have ACME enabled by default.
+
```

Comment:
I think this comment is wrong.  ACME service should be available after CA installation, but not be immediately enabled.  The test procedure executes `ipa-acme-manage enable` which supports this.

Reply:
My understanding is that in topology if we enable ACME on any server,  it should be enabled on all the servers.

So, IMO in topology where one server is CA-less, when converted to CA-full, ACME should be by default enabled.

Reply:
@mrizwan93 there are a few different scenarios here:

1. No CA in deployment at all.  `ipa-ca-install` will install the CA, promoting deployment from CA-less to CA-ful.  ACME service is initially disabled and can be enabled by `ipa-acme-manage`.

2. CA-ful deployment with CA-less replica, ACME enabled.  In this case, `ipa-ca-install` will install the CA role on the previously CA-less replica, and ACME service should be active on that replica automatically.

3. CA-ful deployment with CA-less replica, ACME not enabled.  In this case `ipa-ca-install` will install the CA role on the previously CA-less replica, and ACME service remains disabled on all CA replicas, until an adminstrator enables it.

Do you agree?

Reply:
- [ ] 1, what do you mean by 'NO CA in deployment'? AFAIK, We cannot install ipa master without CA. `ipa-server-install` itself installs CA.

- [ ] 2 is what I tried to implement here

- [ ] 3 agree

Reply:
@mrizwan93 ok, thanks for clarifying.

1. You can indeed install a FreeIPA deployment (first replica via `ipa-server-install`) without a CA.  This is a "CA-less deployment".  This is why I was confused about your comment, where you refer to "CA-less deployment".  When there is a CA-ful deloyment (that is, a deployment which has at least one CA replica), we would refer to "CA-less *replica*", but the deployment is "CA-ful".  I do not think we need a specific test for what happens with ACME when installing first CA server into a CA-less deployment.  But it would not hurt to test it.

2. OK, that sounds good.  So I think the comment / test description needs clarification, to say that (a) deployment is CA-ful, (b) ACME is enabled, (c) promoting a CA-less replica to CA-ful results in ACME immediately enabled on that replica.

3. :+1: